sklearn/utils/tests/test_testing.py

```diff
@@ -491,3 +495,94 @@ def test_check_docstring_parameters():
          'type definition for param: "c " (type definition was "")',
          'sklearn.utils.tests.test_testing.f_check_param_definition There was '
          'no space between the param name and colon ("d:int")'])
+
+
+def func_doc1(y_true, y_pred, sample_weight):
+    """Dummy function for docstring testing.
+
+    Parameters
+    ----------
+    y_true : 1d array-like
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like
+        Estimated targets as returned by a classifier.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    Returns
+    -------
+    fbeta_score : float or array of float, shape = [n_unique_labels]
+        F-beta score of the positive class in binary classification.
+
+    precision : float or array of float, shape = [n_unique_labels]
+        Precision values.
+
+    recall : float or array of float, shape = [n_unique_labels]
+
+    """
+    pass
+
+
+def func_doc2(y_true, y_pred, sample_weight):
+    """Dummy function for docstring testing.
+
+    Parameters
+    ----------
+    y_true : 1d array-like
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like
+        Estimated targets as returned by a classifier.
+
+    sample_weight : array-like of shape = [n_samples], optional
+
+    Returns
+    -------
+    fbeta_score : float or array of float, shape = [n_unique_labels]
+        F-beta score of the positive class in binary classification.
+
+    precision : float or array of float
+        Precision values.
+
+    """
+    pass
+
+
+@if_numpydoc
+def test_assert_consistent_docs():
+    # Test with dummy functions
+    assert_consistent_docs([func_doc1, func_doc2],
+                           include_params=['y_true', 'y_pred'],
+                           include_returns=False)
+    assert_consistent_docs([func_doc1, func_doc2],
+                           exclude_params=['sample_weight'],
+                           include_returns=False)
+
+    # Using NumpyDocString object
+    from numpydoc import docscrape
+
+    doc1 = docscrape.NumpyDocString(inspect.getdoc(func_doc1))
+    doc2 = docscrape.NumpyDocString(inspect.getdoc(func_doc2))
+
+    assert_raises(AssertionError, assert_consistent_docs, [doc1, doc2],
```

Comment:
we should test the message too. Please use `assert_raises_regex` (or `with pytest.raises(AssertionError, match=regex)`)

Reply:
I will do it.